src/main/java/edu/kit/compiler/semantic/NamespaceMapper.java

```diff
@@ -0,0 +1,64 @@
+package edu.kit.compiler.semantic;
+
+import edu.kit.compiler.data.CompilerException;
+import edu.kit.compiler.data.ast_nodes.ClassNode;
+import edu.kit.compiler.data.ast_nodes.MethodNode;
+import lombok.Data;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class NamespaceMapper {
+
+    private final Map<Integer, ClassNamespace> symbolTableMap = new HashMap<>();
+
+    @Data
+    public static class ClassNamespace {
+        private final int classNodeRef;
+        private final Map<Integer, MethodNode.DynamicMethodNode> dynamicMethods = new HashMap<>();
+        private final Map<Integer, MethodNode.StaticMethodNode> staticMethods = new HashMap<>();
+        // Inner symbol table that holds most recent definitions of control flow
+        private final SymbolTable classSymbolTable = new SymbolTable();
```

Comment:
What is this `classSymbolTable` for? Given I use the `getClassNamespace` method, what will I find in this symbol table? The temporary symbol table when going through methods is a already a field in my visitor class, if that was the intention.

Reply:
This would be the per class symbol table but if you already defined this in your Visitor it should be als okay then I remove it here.